pkg/util/containers/providers/cgroup/cgroup_metrics.go

```diff
@@ -182,17 +182,20 @@ func (c ContainerCgroup) SoftMemLimit() (uint64, error) {
 }
 
 // CPU returns the CPU status for this cgroup instance
-// If the cgroup file does not exist then we just log debug return nothing.
+// If the cgroup file does not exist then we return an error; otherwise the metrics
+// struct will be defaulted and set CPU values to 0, leading to misleading results
+// (since CPU is reported cumulatively).
 func (c ContainerCgroup) CPU() (*metrics.ContainerCPUStats, error) {
-	ret := &metrics.ContainerCPUStats{}
 	statfile := c.cgroupFilePath("cpuacct", "cpuacct.stat")
 	f, err := os.Open(statfile)
 	if os.IsNotExist(err) {
 		log.Debugf("Missing cgroup file: %s", statfile)
-		return ret, nil
+		return nil, fmt.Errorf("cgroup file does not exist")
```

Comment:
I think we can remove the debug log and put the `statfile` inside the error

Reply:
thanks! i was thinking the same thing, but then thought i'd leave it for consistency with the other methods. think it's too redundant? i don't feel strongly either way

Reply:
We can keep the `Debug` log for consistency but in any case the error should container the file path, it'll ease investigations when we get non-debug flares.

Reply:
makes sense!